core/arch/arm/kernel/generic_boot.c

```diff
@@ -901,10 +901,10 @@ static void init_external_dt(unsigned long phys_dt)
 		return;
 	}
 
-	if (!core_mmu_add_mapping(MEM_AREA_IO_NSEC, phys_dt, CFG_DTB_MAX_SIZE))
+	if (!core_mmu_add_mapping(MEM_AREA_IO_SEC, phys_dt, CFG_DTB_MAX_SIZE))
```

Comment:
This is not generic.
Platform would need to provide a hint whether this area shall be mapped secure or non-secure.
If it is in secure memory, it is likely the platform has mapped the area.
Then this could be replaced with a test on `phys_dt` and size against `RAM_SEC` (maybe also `RAM_NSEC`). And fallback to `core_mmu_add_mapping()` if not found.

Reply:
I'm currently initializing the TZC380 on my nitrogen6x board as a driver using driver init.
This means that when hitting this code, the TZC380 has not been setup yet and the default for this controller is to configure the whole memory as secure. While the mapping is dependent on the platform, this mapping is no longer needed after the fixup and could be removed, since the memory will be reclaimed by linux anyway. Otherwise I'll have to initialize the TZC380 before `init_external_dt` is hit.

Reply:
Consider platforms using an external DTB in non-secure memory and using non reconfigurable firewalls.

Reply:
Yeah, considering a platform where the firewall blocks secure accesses to non-secure memory and the dtb lies in secure memory, this wouldn't work.
But I think providing function to identify whether the firewall is active is better than a hint, since for my board at least the firewall could be inactive.
Thinking about it, both bits of information are required, whether the firewall is active and whether the dtb lies in secure memory or not.

Reply:
So your DTB needs to be accessed with secure rights before TZC drivers inits and with non-secure rights after TZC driver inits. If think the current P-R will not permit this.

Reply:
Unless security inversion is enabled for the TZC380, non-secure access always implies secure access as well, thats why this PR is working fine for me.